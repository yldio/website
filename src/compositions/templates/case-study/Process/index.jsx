import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import GatsbyImage from 'gatsby-image';

import PageSection from 'components/PageSection';
import { Copy, H2 } from 'components/Typography';

const Process = ({ image, imageAlt, top, main }) => (
  <Fragment>
    <PageSection>
      <H2>The Process</H2>
      <Copy large>{top}</Copy>
      <Copy newLines>{main}</Copy>
    </PageSection>
    <PageSection wide>
      <GatsbyImage {...image.childImageSharp} alt={imageAlt} />
    </PageSection>
  </Fragment>
);

Process.propTypes = {
  main: PropTypes.string.isRequired,
  top: PropTypes.string.isRequired,
  image: PropTypes.shape({
    childImageSharp: PropTypes.shape(GatsbyImage.propTypes).isRequired
  }).isRequired,
  imageAlt: PropTypes.string.isRequired
};

export default Process;
