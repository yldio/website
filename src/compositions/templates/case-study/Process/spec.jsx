import React from 'react';
import { shallow } from 'enzyme';

import Process from './index';

test('success section', () => {
  const props = {
    top: 'Process section top text',
    main: 'Process section main text',
    image: {
      childImageSharp: {
        src: 'process-image.jpg'
      }
    },
    imageAlt: 'A demo of the process'
  };

  const wrapper = shallow(<Process {...props} />);

  expect(wrapper).toMatchSnapshot();
});
