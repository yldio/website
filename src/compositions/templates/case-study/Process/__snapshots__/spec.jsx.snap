// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success section 1`] = `
<React.Fragment>
  <PageSection>
    <Typography__H2>
      The Process
    </Typography__H2>
    <Typography__base
      large={true}
    >
      Process section top text
    </Typography__base>
    <Typography__base
      newLines={true}
    >
      Process section main text
    </Typography__base>
  </PageSection>
  <PageSection
    wide={true}
  >
    <Image
      Tag="div"
      alt="A demo of the process"
      fadeIn={true}
      src="process-image.jpg"
    />
  </PageSection>
</React.Fragment>
`;
